tests/foreman/cli/test_contentviews.py

```diff
@@ -1847,6 +1847,293 @@ def test_cv_publish_version_changes_in_source_env(self):
+
+    def test_cv_subscribe_system_rh(self):
+        """@test: Attempt to subscribe content host to content view that has
+        Red Hat repository assigned to it
+
+        @feature: Content Views
+
+        @assert: Content Host can be subscribed to content view with Red Hat
+        repository
+
+        """
+        self.create_rhel_content()
+        env = make_lifecycle_environment({
+            u'organization-id': self.rhel_content_org['id'],
+        })
+        cv = make_content_view({
+            u'organization-id': self.rhel_content_org['id'],
+        })
+        result = ContentView.add_repository({
+            u'id': cv['id'],
+            u'repository-id': TestContentView.rhel_repo['id'],
+            u'organization-id': TestContentView.rhel_content_org['id'],
+        })
+        self.assertEqual(result.return_code, 0)
+        result = ContentView.info({u'id': cv['id']})
+        self.assertEqual(
+            result.stdout['yum-repositories'][0]['name'],
+            self.rhel_repo_name,
+        )
+        ContentView.publish({u'id': cv['id']})
+        result = ContentView.info({u'id': cv['id']})
+        cvv = result.stdout['versions'][0]
+        result = ContentView.version_promote({
+            u'id': cvv['id'],
+            u'to-lifecycle-environment-id': env['id'],
+        })
+        self.assertEqual(result.return_code, 0)
+
+        result = ContentView.info({u'id': cv['id']})
+        self.assertEqual(result.stdout['content-host-count'], '0')
+
+        make_content_host({
+            u'name': gen_alphanumeric(),
+            u'organization-id': self.rhel_content_org['id'],
+            u'lifecycle-environment-id': env['id'],
+            u'content-view-id': cv['id'],
+        })
+        result = ContentView.info({u'id': cv['id']})
+        self.assertEqual(result.stdout['content-host-count'], '1')
+
+    def test_cv_subscribe_system_rh_spin(self):
+        """@test: Attempt to subscribe content host to filtered content view
+        that has Red Hat repository assigned to it
+
+        @feature: Content Views
+
+        @assert: Content Host can be subscribed to filtered content view with
+        Red Hat repository
+
+        """
+        self.create_rhel_content()
+        env = make_lifecycle_environment({
+            u'organization-id': self.rhel_content_org['id'],
+        })
+        cv = make_content_view({
+            u'organization-id': self.rhel_content_org['id'],
+        })
+        result = ContentView.add_repository({
+            u'id': cv['id'],
+            u'repository-id': TestContentView.rhel_repo['id'],
+            u'organization-id': TestContentView.rhel_content_org['id'],
+        })
+        self.assertEqual(result.return_code, 0)
+        result = ContentView.info({u'id': cv['id']})
+        self.assertEqual(
+            result.stdout['yum-repositories'][0]['name'],
+            self.rhel_repo_name,
+        )
+
+        name = gen_string('alphanumeric')
```

Comment:
If UTF8 is accepted then should use it as it have more complex character representation type.